vendor/github.com/gardener/machine-controller-manager/pkg/test/integration/common/framework.go

```diff
@@ -0,0 +1,882 @@
+	// used only if control cluster is seed
+	mcmDeploymentOrigObj *appsV1.Deployment
+
+	// machineControllerManagerDeploymentName specifies the name of the deployment
+	// running the mc and mcm containers in it.
+	machineControllerManagerDeploymentName = os.Getenv("MACHINE_CONTROLLER_MANAGER_DEPLOYMENT_NAME")
+
+	// names of machineclass resource.
+	testMachineClassResources = []string{"test-mc-v1", "test-mc-v2"}
+
+	// path for v1machineclass yaml file to be used while creating machine resources
+	// name of the machineclass will always be test-mc-v1. overriding the name of machineclass in yaml file
+	// ignored if control cluster is seed cluster
+	v1MachineClassPath = os.Getenv("MACHINECLASS_V1")
+
+	// path for v1machineclass yaml file to be used while upgrading machine deployment
+	// if machineClassV2 is not set then v1MachineClassPath will be used intead for creating test-mc-v2 class
+	// ignored if control cluster if seed cluster
+	v2MachineClassPath = os.Getenv("MACHINECLASS_V2")
+)
+
+type IntegrationTestFramework struct {
+	// Must be rti implementation for the hyperscaler provider.
+	// It is used for checking orphan resources.
+	resourcesTracker helpers.ResourcesTrackerInterface
+
+	// Control cluster resource containing ClientSets for accessing kubernetes resources
+	// And kubeconfig file path for the cluster
+	// initialization is done by SetupBeforeSuite
+	ControlCluster *helpers.Cluster
+
+	// Target cluster resource containing ClientSets for accessing kubernetes resources
+	// And kubeconfig file path for the cluster
+	// initialization is done by SetupBeforeSuite
+	TargetCluster *helpers.Cluster
+
+	// timeout for Eventually to probe kubernetes cluster resources
+	// for machine creation, deletion, machinedeployment update
+	// can be different for different cloud providers
+	timeout time.Duration
+
+	// pollingInterval for Eventually to probe kubernetes cluster resources
+	// for machine creation, deletion, machinedeployment update
+	// can be different for different cloud providers
+	pollingInterval time.Duration
+}
+
+// NewIntegrationTestFramework creates a new IntegrationTestFramework
+// initializing resource tracker implementation.
+// Optially the timeout and polling interval are configurable as optional arguments
+// The default values used for Eventually to probe kubernetes cluster resources is
+// 300 seconds for timeout and  2 seconds for polling interval
+// for machine creation, deletion, machinedeployment update e.t.c.,
+// The first optional argument is the timeoutSeconds
+// The second optional argument is the pollingIntervalSeconds
+func NewIntegrationTestFramework(resourcesTracker helpers.ResourcesTrackerInterface, intervals ...int64) (c *IntegrationTestFramework) {
+	timeout := 300 * time.Second
+	pollingInterval := 2 * time.Second
+	if len(intervals) > 0 {
+		timeout = time.Duration(intervals[0]) * time.Second
+	}
+	if len(intervals) > 1 {
+		pollingInterval = time.Duration(intervals[0]) * time.Second
+	}
```

Comment:
@rkjakeer shouldn't the index be 1 here as pollingInterval will be passed as second parameter?

Reply:
true 👍